app/pages/addFinancialInstitution/WhichIdentificationNumbersPage.scala

```diff
@@ -16,13 +16,28 @@
 
 package pages.addFinancialInstitution
 
-import models.WhichIdentificationNumbers
-import pages.QuestionPage
+import models.{UserAnswers, WhichIdentificationNumbers}
+import pages.{CompanyRegistrationNumberPage, QuestionPage, TrustURNPage}
 import play.api.libs.json.JsPath
 
+import scala.util.Try
+
 case object WhichIdentificationNumbersPage extends QuestionPage[Set[WhichIdentificationNumbers]] {
 
   override def path: JsPath = JsPath \ toString
 
   override def toString: String = "whichIdentificationNumbers"
+
+  def cleanUpUnselectedTINPages(selectedTINs: Set[String], userAnswers: UserAnswers): Try[UserAnswers] = {
```

Comment:
We can still keep the strict typing and don't need to use String here, instead can use `WhichIdentificationNumbers`